code/modules/events/minor/gimmick_flood.dm

```diff
@@ -0,0 +1,70 @@
+/datum/random_event/minor/gimmick_flood
+	name = "Random Flood"
+	disabled = !ASS_JAM
+	weight = 30
+	customization_available = 1
+	var/reagent_type = null
+	var/amount = null
+	var/turf/target = null
+
+	admin_call(var/source)
+		if (..())
+			return
+
+		message_admins("yee")
+
+		if (alert(usr, "Random reagent?", "Random reagent?", "Yes", "No") == "No")
+			var/list/L = list()
+			var/searchFor = input(usr, "Look for a part of the reagent name (or leave blank for all)", "Add reagent") as null|text
+			if(searchFor)
+				for(var/R in childrentypesof(/datum/reagent))
+					if(findtext("[R]", searchFor)) L += R
+			else
+				L = childrentypesof(/datum/reagent)
+
+			if(L.len == 1)
+				reagent_type = L[1]
+			else if(L.len > 1)
+				reagent_type = input(usr,"Select Reagent:","Reagents",null) as null|anything in L
+			else
+				usr.show_text("No reagents matching that name", "red")
+				return
+
+		amount = input(usr,"Amount:","Amount",2000) as null|num
+
+		if (alert(usr, "Target location:", "Target location:", "Random", "My turf") == "My turf")
+			src.target = get_turf(usr)
+
+		src.event_effect(source)
+		return
+
+	event_effect()
+		..()
+
+		if(isnull(src.target))
+			src.target = pick(get_area_turfs(/area/station)) // don't @ me
```

Comment:
could do a random drain, like the sewer is overflowing

Reply:
Good idea, I'll add that as a variant of the event. It could even clog the drain!
Damn, it would be so cool if drains used reagent pipes.